class CompletableFutureTest
!!!189422.java!!!	setUp() : void
    Retrofit retrofit = new Retrofit.Builder()
        .baseUrl(server.url("/"))
        .addConverterFactory(new StringConverterFactory())
        .addCallAdapterFactory(Java8CallAdapterFactory.create())
        .build();
    service = retrofit.create(Service.class);
!!!189550.java!!!	bodySuccess200() : void
    server.enqueue(new MockResponse().setBody("Hi"));

    CompletableFuture<String> future = service.body();
    assertThat(future.get()).isEqualTo("Hi");
!!!189678.java!!!	bodySuccess404() : void
    server.enqueue(new MockResponse().setResponseCode(404));

    CompletableFuture<String> future = service.body();
    try {
      future.get();
      fail();
    } catch (ExecutionException e) {
      assertThat(e.getCause())
          .isInstanceOf(HttpException.class) // Required for backwards compatibility.
          .isInstanceOf(retrofit2.HttpException.class)
          .hasMessage("HTTP 404 Client Error");
    }
!!!189806.java!!!	bodyFailure() : void
    server.enqueue(new MockResponse().setSocketPolicy(DISCONNECT_AFTER_REQUEST));

    CompletableFuture<String> future = service.body();
    try {
      future.get();
      fail();
    } catch (ExecutionException e) {
      assertThat(e.getCause()).isInstanceOf(IOException.class);
    }
!!!189934.java!!!	responseSuccess200() : void
    server.enqueue(new MockResponse().setBody("Hi"));

    CompletableFuture<Response<String>> future = service.response();
    Response<String> response = future.get();
    assertThat(response.isSuccessful()).isTrue();
    assertThat(response.body()).isEqualTo("Hi");
!!!190062.java!!!	responseSuccess404() : void
    server.enqueue(new MockResponse().setResponseCode(404).setBody("Hi"));

    CompletableFuture<Response<String>> future = service.response();
    Response<String> response = future.get();
    assertThat(response.isSuccessful()).isFalse();
    assertThat(response.errorBody().string()).isEqualTo("Hi");
!!!190190.java!!!	responseFailure() : void
    server.enqueue(new MockResponse().setSocketPolicy(DISCONNECT_AFTER_REQUEST));

    CompletableFuture<Response<String>> future = service.response();
    try {
      future.get();
      fail();
    } catch (ExecutionException e) {
      assertThat(e.getCause()).isInstanceOf(IOException.class);
    }
